### PR TITLE
fix screen position for server / login / forgot password & mfa

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Fixes the scrolling of the contents of the following screens on iPhone, iPad & Android as part of broken window fixes

- Server
- Login
- Forgot password
- MFA

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56150

Fixes: #8291

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/screens/mfa/index.tsx [hunk 1]
```diff
@@ -156,6 +142,16 @@ const MFA = ({componentId, config, goToHome, license, loginId, password, serverD
...
 
+    useEffect(() => {
+        requestAnimationFrame(() => {
+            let height = keyboard / 2;
+            if (height < 80) {
+                height = 0;
+            }
+            keyboardAwareRef.current?.scrollToPosition(0, height);
+        });
+    }, [keyboard, keyboardAwareRef]);
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/screens/mfa/index.tsx:153]: Sure that seems better.. will do

**Review by larkox — APPROVED** [review]
Beautiful!

File: app/screens/forgot_password/index.tsx [hunk 2]
```diff
@@ -270,7 +256,7 @@ const ForgotPassword = ({componentId, serverUrl, theme}: Props) => {
         const unsubscribe = Navigation.events().registerComponentListener(listener, componentId);
 
         return () => unsubscribe.remove();
-    }, [dimensions]);
+    }, [componentId, dimensions, translateX]);
```
Comment (rahimrahman) [inline app/screens/forgot_password/index.tsx:259]:
```suggestion
    // eslint-disable-next-line react-hooks/exhaustive-deps treat translateX.value like useRef
    }, [componentId, dimensions.width]);
```

Comment (rahimrahman) [reply app/screens/forgot_password/index.tsx:259]: Ref:

https://docs.swmansion.com/react-native-reanimated/docs/core/useSharedValue/#remarks

Comment (enahum) [reply app/screens/forgot_password/index.tsx:259]: Is not being modified, but ok

**Comment by enahum** [comment]
@matthewbirtch is not related to this PR but here is the thing, I can tell you for a fact that enforcing the MFA before the user has logged in, on mobile is not implemented, so the server will return the error and the mobile app will not handle it by requesting the MFA from the user. so you can try it if you enable MFA before logging in.

Yes, we should create a ticket when MFA is enforced after the user has logged in.

**Comment by enahum** [comment]
https://mattermost.atlassian.net/browse/MM-61834

**Comment by matthewbirtch** [comment]
> @matthewbirtch is not related to this PR but here is the thing, I can tell you for a fact that enforcing the MFA before the user has logged in, on mobile is not implemented, so the server will return the error and the mobile app will not handle it by requesting the MFA from the user. so you can try it if you enable MFA before logging in.
> 
> Yes, we should create a ticket when MFA is enforced after the user has logged in.

Yep, confirmed I was able to test the MFA token screen now after going through he MFA setup on webapp first. 

I'll create a ticket for adding this MFA setup screen to the mobile flow.

**Pull request merged**